PatternPal/PatternPal/Services/StepByStepService.cs

```diff
@@ -1,41 +1,154 @@
-﻿using PatternPal.Core.StepByStep;
-using PatternPal.SyntaxTree;
-using PatternPal.SyntaxTree.Abstractions.Entities;
+﻿#region
+
+using PatternPal.Core.Recognizers;
+using PatternPal.Core.StepByStep;
+
+#endregion
 
 namespace PatternPal.Services;
 
 public class StepByStepService : Protos.StepByStepService.StepByStepServiceBase
 {
+    /// <inheritdoc />
     public override Task< GetInstructionSetsResponse > GetInstructionSets(
         GetInstructionSetsRequest request,
         ServerCallContext context)
     {
-        throw new NotImplementedException();
+        GetInstructionSetsResponse response = new();
+        
+        IDictionary<Recognizer, IRecognizer> sbsDictionary = 
+            RecognizerRunner.SupportedRecognizers;
+
+        // Go over all the recognizers supported by the runner and remove those that do not implement
+        // the GenerateStepsList function.
+        foreach (KeyValuePair<Recognizer, IRecognizer> entry in sbsDictionary)
+        {
+            try
+            {
+                entry.Value.GenerateStepsList();
+            }
+            catch
+            {
+                sbsDictionary.Remove(entry.Key);
+            }
+        }
+
+        response.Recognizers.Add(sbsDictionary.Keys);
```

Comment:
```suggestion
        foreach (
        KeyValuePair<Recognizer, IRecognizer> entry in RecognizerRunner.SupportedRecognizers)
        {
            try
            {
                entry.Value.GenerateStepsList();
            }
            catch
            {
                continue;
            }
            
            response.Recognizers.Add(entry.Key);
        }
```

Reply:
I have done this differently now with the suggestion.